compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1864,6 +1864,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         | ty::Str
                         | ty::Projection(_)
                         | ty::Param(_) => format!("{deref_ty}"),
+                        // we need to test something like  <&[_]>::len
+                        // and Vec::function();
+                        // <&[_]>::len doesn't need an extra "<>" between
+                        // but for Adt type like Vec::function()
+                        // we would suggest <[_]>::function();
+                        ty::Slice(_) if self.tcx.sess.source_map().span_wrapped_by_angle_bracket(ty.span)  => format!("{deref_ty}"),
                         _ => format!("<{deref_ty}>"),
```

Comment:
I don't think it should just be `ty::Slice` as we recommend `<{deref_ty}>` to all types below. So we should stop recommending `<{deref_ty}>` no matter what `deref_ty` is if there are angle brackets already.

Reply:
yes. You're right. We should not recommend <{deref_ty}> only for ty::Slice.
And the change was pushed.
Thank you for the suggestions.